### ISSUE TITLE
Export all notes and tags in CSV or JSON format

### ISSUE THREAD

**Issue by debritto** [ezra-bible-app/ezra-bible-app#469]
Hi folks,
Is it possible to create a option to export all notes and tags in a CSV or JSON format?

All that I need is to export all this data:
```
SELECT   "BibleBooks"."shortTitle" as BibleBook,  "VerseReferences"."chapter",  "VerseReferences"."verseNr",  "Tags"."title" as Tag,   "Notes"."text" as Note FROM   "BibleBooks" INNER JOIN "VerseReferences"  ON "BibleBooks"."id" = "VerseReferences"."bibleBookId" INNER JOIN "VerseTags"  ON "VerseReferences"."id" = "VerseTags"."verseReferenceId" INNER JOIN "Tags"  ON "VerseTags"."tagId" = "Tags"."id" INNER JOIN "Notes"  ON "VerseReferences"."id" = "Notes"."verseReferenceId" 
```

My plan is to use this information to create some charts and make some analysis.
I am looking forward to hearing from you.
Blessings in Jesus name.

**Comment by debritto** [comment]
Oops. I forgot to say that my students will use the "Export all" option and they aren't advanced users. In a matter of fact, I believe that most users that could use this feature will be plain users. ;)

**Comment by tobias-klein** [comment]
> Just another thing. Do you think that is possible to export the content of the related Bible verses to (At least to Bible translations in Public Domain)?

Yes, should be possible. We will simply show a disclaimer to the user based on the modules with Copyright. We are doing the same in case of #96.

**Comment by zhuiks** [comment]
Here are my thoughts:
1. Regarding UI, I think the most natural/minimalistic would be to add "Export All" under dropdown next to "Export to Word" button: 
![Screenshot from 2021-10-15 20-14-25](https://user-images.githubusercontent.com/3584119/137568136-5f6e1e9f-755c-45ea-8d21-dca0134aeaa2.png)
2. I think the CSV format is more universal than JSON for non-technical users. It can be open with any spreadsheets program.
3. I definitely agree with the "Export All" concept - one click to dump all user-generated data. I'm a bit more skeptical about including Bible verses content. It's not "user-generated". From my experience, end-users either care too much about copyright or don't care at all (with or without disclaimer dialog). However, I guess adding Bible text to dump for Public Domain or some other open licenses. As it is publicly available anyway.

I think it would be possible to implement for the 1.4 version.

**Comment by tobias-klein** [comment]
I like your proposal @zhuiks. And I agree that CSV is easier to handle for the average user. However, the question is what's the purpose of this feature. If the export shall be used for further computer-based data processing (that's what I understood from @debritto), then a format like JSON may be more suitable, because you can "imitate" multiple database tables via that format (like having a list of tags and then along with that a list of tag assignments that reference those tags). Also, if we agree on a certain JSON format then this could be discussed with other frontend's developers and we may be able to establish a common format for export (and potentially also import). Furthermore, if the purpose is rather technical then for me the question is whether this needs to be accessible via a standard menu option or can also be a bit hidden, because it's very likely not used very often ...
